src/stores/web3.ts

```diff
@@ -45,9 +45,15 @@ export default class Web3Store {
 
     this.injectedAvailable = !!(ethereum || web3);
     if (metamask) {
-      metamask.isApproved().then(approved => {
-        this.updateInjected(approved ? ethereum : null, approved);
-      });
+      setTimeout(() =>{
+        if (metamask.isEnabled()) {
+          this.updateInjected(ethereum);
+        } else {
+          metamask.isApproved().then(approved => {
+            this.updateInjected(approved ? ethereum : null, approved);
+          });
+        }
+      }, 500);
```

Comment:
shitty solution with setTimeout. `isEnabled` is set async by metamask and sometimes our code may execute before that. Tried listening for `ethereumprovider` event ([link](https://github.com/MetaMask/metamask-extension/pull/4703/files#diff-205df9c93990ff0f805d90215a20cda4R152)) but didn't work.